evcouplings/complex/protocol.py

```diff
@@ -111,27 +315,373 @@ def _modify_segments(seg_list, seg_prefix):
+                           target_sequence,
+                           alignment_file,
+                           identity_threshold):
+        # read in annotation to a file and rename the appropriate column
+        id_to_organism = read_species_annotation_table(annotations_file)
+
+        # This dictionary maps identifiers used to calculate pairing
+        # to identifiers used to index into the sequence alignment
+        # Make sure the target sequence is included in this dictionary
+        # TODO: fix this so that we don"t assume target sequence is the first sequence
+        id_to_header = {x: [x] for x in id_to_organism.id}
+        id_to_header[target_sequence] = [Alignment.from_file(open(alignment_file)).ids[0]]
+
+        # read identity file, then do a quick groupby / max
+        similarities = pd.read_csv(identities_file)
+        most_similar_in_species = most_similar_by_organism(similarities, id_to_organism)
+
+        paralogs = find_paralogs(
+            target_sequence, id_to_organism,
+            similarities, identity_threshold
+        )
+
+        species_to_bestreciprocal = filter_best_reciprocal(
+            alignment_file, paralogs, most_similar_in_species
+        )
+
+        return species_to_bestreciprocal, id_to_header
+
+    # load the information about each monomer alignment
+    species_to_bestreciprocal_1, id_to_header_1 = _load_monomer_info(
+        kwargs["first_annotation_file"],
+        kwargs["first_identities_file"],
+        kwargs["first_focus_sequence"],
+        kwargs["first_alignment_file"],
+        kwargs["paralog_identity_threshold"]
+    )
+
+    species_to_bestreciprocal_2, id_to_header_2 = _load_monomer_info(
+        kwargs["second_annotation_file"],
+        kwargs["second_identities_file"],
+        kwargs["second_focus_sequence"],
+        kwargs["second_alignment_file"],
+        kwargs["paralog_identity_threshold"]
+    )
+
+    # merge the two dataframes to get all species found in 
+    # both alignments
+    species_intersection = species_to_bestreciprocal_1.merge(
+        species_to_bestreciprocal_2,
+        how="inner",  # takes the intersection
+        on="species",  # merges on species identifiers
+        suffixes=("_1", "_2")
+    )
+
+    raw_alignment_file = prefix + "_raw.fasta"
+
+    target_seq_id, target_seq_index = write_concatenated_alignment(
+        species_intersection,
+        id_to_header_1, id_to_header_2,
+        kwargs["first_alignment_file"],
+        kwargs["second_alignment_file"],
+        kwargs["first_focus_sequence"],
+        kwargs["second_focus_sequence"],
+        raw_alignment_file
+    )
+
+    # filter the alignment
+    raw_ali = Alignment.from_file(open(raw_alignment_file))
+    aln_outcfg, _ = modify_alignment(
+        raw_ali,
+        target_seq_index,
+        target_seq_id,
+        kwargs["first_region_start"],
+        **kwargs
+    )
+
+    def _modify_segments(seg_list, seg_prefix):
```

Comment:
move all this redundant logic in 3 protocols to own function (see comment above)